x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceService.java

```diff
@@ -69,21 +69,39 @@ public class ElasticInferenceService extends SenderService {
 
     private final ElasticInferenceServiceComponents elasticInferenceServiceComponents;
 
-    private static final EnumSet<TaskType> supportedTaskTypes = EnumSet.of(TaskType.SPARSE_EMBEDDING, TaskType.COMPLETION);
+    private static final EnumSet<TaskType> IMPLEMENTED_TASK_TYPES = EnumSet.of(TaskType.SPARSE_EMBEDDING, TaskType.COMPLETION);
     private static final String SERVICE_NAME = "Elastic";
 
+    private final Configuration configuration;
+    private final EnumSet<TaskType> enabledTaskTypes;
+
     public ElasticInferenceService(
         HttpRequestSender.Factory factory,
         ServiceComponents serviceComponents,
         ElasticInferenceServiceComponents elasticInferenceServiceComponents
     ) {
         super(factory, serviceComponents);
         this.elasticInferenceServiceComponents = elasticInferenceServiceComponents;
+        enabledTaskTypes = enabledTaskTypes(this.elasticInferenceServiceComponents.acl());
+        configuration = new Configuration(enabledTaskTypes);
+    }
+
+    private static EnumSet<TaskType> enabledTaskTypes(ElasticInferenceServiceACL acl) {
+        var implementedTaskTypes = EnumSet.copyOf(IMPLEMENTED_TASK_TYPES);
+        implementedTaskTypes.retainAll(acl.enabledTaskTypes());
+        return implementedTaskTypes;
     }
 
     @Override
     public Set<TaskType> supportedStreamingTasks() {
-        return COMPLETION_ONLY;
+        var enabledStreamingTaskTypes = EnumSet.of(TaskType.COMPLETION);
+        enabledStreamingTaskTypes.retainAll(enabledTaskTypes);
+
+        if (enabledStreamingTaskTypes.isEmpty() == false) {
```

Comment:
If there are no enabled task types we won't add `any` since we don't want to support anything.